### ISSUE TITLE
Adding model utilities

### ISSUE THREAD

**Issue by shan18** [shan18/TensorNet#3]
Include support for more loss functions and optimizers.

**Comment by shan18** [comment]
New Loss functions added in 960ce345cc62a8a3979b481eae42a75af4c4eef3:
- Binary Cross-Entropy
- SSIM
- Mean Square Error

**Comment by shan18** [comment]
Dice loss added in edbea354a3782d2e20f263a5f7dea19f2dc16d03

**Comment by shan18** [comment]
New loss functions added in 079019e860b9d35a2431c7216f200cec41ed3694.

**Comment by shan18** [comment]
Adam optimizer added in 757416046de1329fb295238902d043fb8afb6801
